static/js/shared/util.ts

```diff
@@ -56,29 +56,38 @@ export function randDomId(): string {
     .substr(2, 10);
 }
 
+/**
+ * Downloads a file under a given filename.
+ * @param filename name to download the file to
+ * @param file the file to download
+ */
+export function downloadFile(fileName: string, file: Blob | File): void {
+  const link = document.createElement("a");
+  const url = window.URL.createObjectURL(file);
+  link.setAttribute("href", url);
+  link.setAttribute("download", fileName);
+  link.onclick = () => {
+    setTimeout(() => window.URL.revokeObjectURL(url));
+  };
+  link.click();
+  link.remove();
+}
+
 /**
  * Saves csv to filename.
  * @param {filename} string
  * @param {contents} string
  * @return void
  */
 export function saveToFile(filename: string, contents: string): void {
-  let mimeType = "text/plan";
+  let mimeType = "text/plain";
```

Comment:
wonder what issue this has now?

Reply:
surprisingly, no issues .. but "text/plan" is not a mimeType